multicluster/cmd/multicluster-controller/clusterset_webhook.go

```diff
@@ -57,6 +63,22 @@ func (v *clusterSetValidator) Handle(ctx context.Context, req admission.Request)
 			klog.ErrorS(err, "the field 'clusterID' is immutable", "ClusterSet", klog.KObj(clusterSet))
 			return admission.Denied("the field 'clusterID' is immutable")
 		}
+
+		if isMemberListChanged(oldClusterSet.Spec.Members, clusterSet.Spec.Members) {
+			// Don't allow general user to update member list
+			ui := req.UserInfo
+			_, saName, err := serviceaccount.SplitUsername(ui.Username)
+			if err != nil {
+				klog.ErrorS(err, "Error getting ServiceAccount name", "ClusterSet", req.Namespace+"/"+req.Name)
+				return admission.Errored(http.StatusBadRequest, err)
+			}
+			isMCServiceAccount := saName == serviceAccount
+			err = fmt.Errorf("member list can only be updated by antrea multicluster controller")
```

Comment:
This line should be moved into the `if !isMCServiceAccount {` block.

Reply:
Fixed